helix-gateway/src/main/java/org/apache/helix/gateway/channel/HelixGatewayServicePollModeChannel.java

```diff
@@ -20,38 +20,195 @@
+          pushClientEventToGatewayManager(_manager,
+              StateTransitionMessageTranslateUtil.translateCurrentStateDiffToInitConnectEvent(clusterName, instanceName,
+                  currentStateDiff.containsKey(clusterName) ? currentStateDiff.get(clusterName).get(instanceName)
+                      : new HashMap<>()));
+        }
+        _livenessResults.computeIfAbsent(clusterName, k -> new HashMap<>()).put(instanceName, liveness);
+      }
+    }
   }
 
   @Override
   public void sendStateChangeRequests(String instanceName,
       HelixGatewayServiceOuterClass.ShardChangeRequests shardChangeRequests) {
+    switch (_shardStateChannelType) {
+      case FILE:
+        // we are periodically writing to the file, so no need to write here.
+        break;
+      default:
+        throw new NotImplementedException("Only support file based channel for now");
+    }
   }
 
   @Override
   public void start() throws IOException {
+    logger.info("Starting Helix Gateway Service Poll Mode Channel...");
+    final Runnable fetchUpdatesTask = new Runnable() {
+      @Override
+      public void run() {
+        fetchUpdates();
+      }
+    };
+    _scheduler.scheduleAtFixedRate(fetchUpdatesTask, _config.getPollStartDelaySec(),  // init delay
+        _config.getPollIntervalSec(),            //  poll interval
+        TimeUnit.SECONDS);
+    scheduleTargetStateUpdateTask();
+  }
 
+  void scheduleTargetStateUpdateTask() {
+    if (_shardStateChannelType == GatewayServiceChannelConfig.ChannelType.FILE) {
+      final Runnable writeTargetStateTask = new Runnable() {
+        @Override
+        public void run() {
+          flushAssignmentToFile(_manager.serializeTargetState(), _targetStateFilePath);
+        }
+      };
+      _scheduler.scheduleAtFixedRate(writeTargetStateTask, _config.getPollStartDelaySec(),  // init delay
+          _config.getTargetFileUpdateIntervalSec(),            //  poll interval
+          TimeUnit.SECONDS);
+    }
   }
 
   @Override
   public void stop() {
-
+    logger.info("Stopping Helix Gateway Service Poll Mode Channel...");
+    // Shutdown the scheduler gracefully when done (e.g., on app termination)
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+      _scheduler.shutdown();
+      try {
+        if (!_scheduler.awaitTermination(1, TimeUnit.MINUTES)) {
+          _scheduler.shutdownNow();
+        }
+      } catch (InterruptedException e) {
+        _scheduler.shutdownNow();
+      }
+    }));
   }
 
   @Override
-  public void closeConnectionWithError(String instanceName, String reason) {
-
+  public void closeConnectionWithError(String clusterName, String instanceName, String reason) {
   }
 
   @Override
-  public void completeConnection(String instanceName) {
+  public void completeConnection(String clusterName, String instanceName) {
+  }
+
+  /**
+   * Read current state
```

Comment:
Make an explanation here. Now we just support file based. gRPC server based will come later. Also create an issue to track here.